File: src/scraper/twitch.py
import json
import os
import requests
import logging

BROADCASTER_ID = "29722828"

# Configure logging
logging.basicConfig(
    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
)
logger = logging.getLogger(__name__)


def get_access_token():
    url = "https://id.twitch.tv/oauth2/token"
    params = {
        "client_id": os.getenv("TWITCH_CLIENT_ID"),
        "client_secret": os.getenv("TWITCH_CLIENT_SECRET"),
        "grant_type": "client_credentials",
    }
    response = requests.post(url, params=params)
    response.raise_for_status()
    response = response.json()

    try:
        return response["access_token"]
    except KeyError as e:
        logger.error(f"Access token not found in response: {response}")
        raise e


def get_emotes(access_token):
    headers = {
        "Authorization": f"Bearer {access_token}",
        "Client-Id": os.getenv("TWITCH_CLIENT_ID"),
    }

    params = {
        "broadcaster_id": BROADCASTER_ID,
    }

    response = requests.get(
        "https://api.twitch.tv/helix/chat/emotes", params=params, headers=headers
    )
    response.raise_for_status()

    return response.json()

File: functions/scraper/twitch.js
const axios = require("axios");
const logger = require("../utils/logger");

const BROADCASTER_ID = "29722828";

const getAccessToken = async () => {
  const url = "https://id.twitch.tv/oauth2/token";
  const params = {
    client_id: process.env.TWITCH_CLIENT_ID,
    client_secret: process.env.TWITCH_CLIENT_SECRET,
    grant_type: "client_credentials",
  };

  try {
    const response = await axios.post(url, null, { params });
    return response.data.access_token;
  } catch (error) {
    logger.error(
      `Failed to get access token: ${
        error.response ? error.response.data : error.message
      }`
    );
    throw error;
  }
};

const getEmotes = async (accessToken) => {
  const url = "https://api.twitch.tv/helix/chat/emotes";
  const headers = {
    "Authorization": `Bearer ${accessToken}`,
    "Client-Id": process.env.TWITCH_CLIENT_ID,
  };
  const params = {
    broadcaster_id: BROADCASTER_ID,
  };

  try {
    const response = await axios.get(url, { headers, params });
    return response.data;
  } catch (error) {
    logger.error(
      `Failed to get emotes: ${
        error.response ? error.response.data : error.message
      }`
    );
    throw error;
  }
};

module.exports = { getAccessToken, getEmotes };
